ramips: add support for read/write uboot env to Asus RX-AX53U

Djfe opened the pull request:
Add support for read/writing uboot env by renaming the second partition
to its stock label "nvram" and remove the deemed unnecessary
"read-only". Split the first partition "u-boot" in two, in order
to allow `fw_setenv` safe write-access to the uboot environment
variables.

This implements hauke's request from [1].
Based on the patch provided by Shiji Yang.

[1] https://github.com/openwrt/openwrt/pull/10400#discussion_r945153224

Co-Authored-By: Shiji Yang <yangshiji66@qq.com>
Signed-off-by: Felix Baumann <felix.bau@gmx.de>

Znevna commented:
Nice, thank you, I've had similar edits done locally, but I've specified the number of sectors (1) too, I don't know the proper value to put there, in the other PR it's mentioned a value of 3 sectors, while the space for that partition is of three sectors, but if u-boot only use one..?
https://github.com/ARM-software/u-boot/blob/master/tools/env/fw_env.config#L5

BlackMark29A commented:
I checked how much data the u-boot `saveenv` command writes to flash and it's exactly `0x20000`. So this router is configured to only use 1 sector and the config in this PR should be correct.

Djfe commented:
Did anyone actually check whether `fw_setenv` works and changes (for example to "bootdelay" get applied and survive a reboot?
Apart from that, this is definitely ready to be merged :)

BlackMark29A commented:
I checked `fw_setenv` and the changes are persistent across reboots.

Ansuel commented:
@Djfe looks ok to me just we can put nickname in the tested by tag but only real name and email. If they can't be provided I need to ask you to drop them.

Ansuel commented:
Also would be good to split this to 2 different commits one for uboot-envtools and the other for the ramips target.

Djfe committed:
ramips: add support for read/write uboot env to Asus RX-AX53U

Add support for read/writing uboot env by renaming the second partition
to its stock label "nvram" and remove the deemed unnecessary
"read-only". Split the first partition "u-boot" in two, in order
to allow `fw_setenv` safe write-access to the uboot environment
variables.

This implements hauke's request from [1].
Based on the patch provided by Shiji Yang.

[1] https://github.com/openwrt/openwrt/pull/10400#discussion_r945153224

Co-Authored-By: Shiji Yang <yangshiji66@qq.com>
Signed-off-by: Felix Baumann <felix.bau@gmx.de>
[ improve commit title and description, fix some whitespace problem ]
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Djfe committed:
uboot-envtools: add support for ramips Asus RX-AX53U

Adds uboot-envtools support for ramips Asus RX-AX53U now that partition
can be correctly read.

Signed-off-by: Felix Baumann <felix.bau@gmx.de>
[ improve commit title and description ]
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Djfe commented:
Done :)
@BlackMark29A if you like to be credited in these commits, I would need your real name and email.
If you don't care, that's fine as well. Thanks for getting this shipped.

This is done, we don't have to wait for Mark, I think. (He had 6 hours to reply already)
If you actually see a reply of his after mine, I kindly ask you to wait a bit before merging.

BlackMark29A commented:
Yeah, I don't care about my real name being included. No need to wait.

Ansuel commented:
@Djfe there is still something to polish like the wrong tag in the second commit title and something to improve. I can fix them while I merge this. Just tell me if it's ok for you!

Djfe commented:
sure, go right ahead :)

Ansuel commented:
@Djfe why the nvram partition have the read-only dropped?

Djfe commented:
It's only used to store configuration for the stock firmware (some kind of csv-like key-value store). I just adopted the oppinion of @BlackMark29A
His oppinion was that it doesn't make sense to make non-volatile-ram non-writeable.
It was set read-only when it was still believed to be the uboot-environment in the original commit.

https://github.com/openwrt/openwrt/pull/10400#discussion_r1056977315

You probably know what's best for such a partition.

Ansuel commented:
@Djfe well it's just a change not related to the commit description so I would keep it read-only. Thanks for the explaination. I'm fixing this last thing and I'm merging this.

Djfe commented:
the first commit still says 'and remove the deemed unnecessary
"read-only".'
I only just noticed that 😅

Ansuel commented:
@Djfe i was referring to uboot-env not to nvmem but it's ok... anyway for next contribution as you can see commit description should be imperative instead of descriptive... aka you instruct the change to do and why
